resources/lib/database.py

```diff
@@ -0,0 +1,257 @@
+        result = cur.fetchone()
+        if default_value is not None:
+            data_type = type(default_value)
+        elif data_type is None:
+            data_type = str
+        return common.convert_from_string(result[0], data_type) if result else default_value
+
+    @sql_connect()
+    def get_values(self, key, default_value=None, table=TABLE_APP_CONF):
+        """
+        Get multiple values from database
+        :param key: The key to get the values
+        :param default_value: When key do not exist return this default value
+        :param table: Table map
+        :return: The values (type string) in a list of tuple
+        """
+        table_name = next(iter(table))
+        table_columns = next(iter(table.values()))
+        query = 'SELECT {} FROM {} WHERE {} = ?'.format(table_columns[1],
+                                                        table_name,
+                                                        table_columns[0])
+        cur = self._execute_query(query, (key,))
+        result = cur.fetchall()
+        return result if result else default_value
+
+    @sql_connect()
+    def set_value(self, key, value, table=TABLE_APP_CONF):
+        """
+        Store a single value to database
+        :param key: The key to store the value
+        :param value: Value to save
+        :param table: Table map
+        """
+        table_name = next(iter(table))
+        table_columns = next(iter(table.values()))
+        # Update or insert approach, if there is no updated row then insert new one (no id changes)
+        update_query = 'UPDATE {} SET {} = ? WHERE {} = ?'.format(table_name,
+                                                                  table_columns[1],
+                                                                  table_columns[0])
+        value = common.convert_to_string(value)
+        cur = self._execute_query(update_query, (value, key))
+        if cur.rowcount == 0:
+            insert_query = 'INSERT INTO {} ({}, {}) VALUES (?, ?)'\
+                .format(table_name, table_columns[0], table_columns[1])
+            self._execute_non_query(insert_query, (key, value))
+
+    @sql_connect()
+    def delete_key(self, key, table=TABLE_APP_CONF):
+        """
+        Delete a key record from database
+        :param key: The key to delete
+        :param table: Table map
+        :return: Number of deleted entries
+        """
+        table_name = next(iter(table))
+        table_columns = next(iter(table.values()))
+        query = 'DELETE FROM {} WHERE {} = ?'.format(table_name, table_columns[0])
+        cur = self._execute_query(query, (key,))
+        return cur.rowcount
+
+    def __del__(self):
+        if self.conn:
+            self.conn.close()
+
+
+class NFDatabase(BaseSQLiteDatabase):
+    def _get_active_guid_profile(self):
+        query = 'SELECT Guid FROM Profiles WHERE IsActive = 1'
+        cur = self._execute_query(query)
+        return cur.fetchone()[0]
```

Comment:
With the template database just copied, I have an error at this line, and can't list my profiles.

See log https://paste.kodi.tv/puvuzanebu.kodi

Reply:
no profiles appears to have been saved to db, it is strange
did you delete my personal profile data from the db? being no longer related it can cause problems

Reply:
No profile on the db, no credential asked.

The db is the last you pushed, with no profile inside.
I also tried to removed my settings.xml to enfoce credential asking, without success.

If it helps, here's a full debug log when I totally remove the userdata\addon_data\plugin.video.netflix folder.
https://paste.kodi.tv/oyoheyapex.kodi

Reply:
And a full debug log after upgrading (before I delete the folder
https://paste.kodi.tv/xekubosujo.kodi

Reply:
the main problem seem that the Session table has disappeared
SQLite error no such table: Session

Reply:
There's also this after the session is initialised in the "upgrade" log

Login prefetch: No stored credentials are available